src/vmemcache.c

```diff
@@ -69,6 +173,61 @@ vmemcache_newU(const char *dir, size_t max_size, size_t fragment_size,
 void
 vmemcache_delete(VMEMcache *cache)
 {
+	while (vmemcache_evict(cache, NULL, 0) == 0)
+		;
+	repl_p_destroy(&cache->repl);
+	ravl_delete(cache->index);
+	vmcache_heap_destroy(cache->heap);
+	munmap(cache->addr, cache->size);
+	Free(cache);
+}
+
+/*
+ * vmemcache_populate_fragments -- (internal) populate fragments
+ */
+static void
+vmemcache_populate_fragments(struct cache_entry *entry,
+				const char *value, size_t value_size)
+{
+	struct heap_entry he;
+	size_t size_left = value_size;
+
+	VEC_FOREACH(he, &entry->value.fragments) {
+		if (he.size <= size_left) {
+			memcpy(he.ptr, value, he.size);
+			value += he.size;
+			size_left -= he.size;
+			if (size_left == 0)
+				break;
+		} else {
+			memcpy(he.ptr, value, size_left);
+			break;
+		}
+	}
+}
+
+/*
+ * vmemcache_entry_delete -- delete the vmemcache entry
+ */
+static int
+vmemcache_entry_delete(VMEMcache *cache, struct cache_entry *entry)
+{
+	struct heap_entry he;
+
+	util_mutex_lock(&entry->value.lock);
```

Comment:
if refcount == 0 then this is owned exclusively by this thread. Don't need to grab this lock.

Reply:
Removed